File: hw14MongoDBAndGUI/Python/SchoolSystem/controller/teacher_controller.py
from controller.database import Database

class TeacherController:

    def __init__(self):
        self.database = Database()

    def save_teacher(self, teacher):

        self.database.teachers.insert_one({
            "id": teacher.teacher_id,
            "name": teacher.name,
            "subject": teacher.subject
        })

    def search_teacher(self, teacher_id):
        query = self.database.teachers.find_one({"id": teacher_id})
        return query
        

    def update_teacher(self, teacher):

        self.database.teachers.update_one(
            {"id": teacher.teacher_id},
            {
                "$set": {
                    "name": teacher.name,
                    "subject": teacher.subject
                }
            }
        )

    def delete_teacher(self, teacher_id):

        self.database.students.delete_one({"id": teacher_id})

File: hw14MongoDBAndGUI/JavaScript/SchoolSystem/controllers/teacherController.js
const mongo = require("../database/mongo");

class TeacherController {

    async saveTeacher(teacher) {

        const db = await mongo.getDb();

        await db
            .collection("teachers")
            .insertOne(teacher);

        console.log("Teacher Saved");
    }

    async searchTeacher(id) {

        const db = await mongo.getDb();

        return await db
            .collection("teachers")
            .findOne({ id });
    }

    async updateTeacher(teacher) {

        const db = await mongo.getDb();

        await db
            .collection("teachers")
            .updateOne(
                { id: teacher.id },
                {
                    $set: {
                        name: teacher.name,
                        subject: teacher.subject
                    }
                }
            );
    }

    async deleteTeacher(id) {

        const db = await mongo.getDb();

        await db
            .collection("teachers")
            .deleteOne({ id });
    }
}

module.exports = TeacherController;